gitlab/resource_gitlab_project.go

```diff
@@ -301,6 +317,22 @@ func resourceGitlabProjectCreate(d *schema.ResourceData, meta interface{}) error
 		options.ImportURL = gitlab.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("template_name"); ok {
+		options.TemplateName = gitlab.String(v.(string))
```

Comment:
Where does `options.TemplateName` come from? It's not in the `CreateProjectOptions` struct or `resourceGitlabProjectSetToState`, does it need to be declared somewhere?

Reply:
In the PR I upgraded go gitlab package to 0.37.0.
You could found here https://github.com/xanzy/go-gitlab/blob/v0.37.0/projects.go#L505

Reply:
I _think_ `options.TemplateName` reads from the `options` struct in resource_gitlab_project.go :277? In which case the attribute needs to be set there

Reply:
I reads from `d ResourceData`, template options doesn't have default values (like `initialize_with_readme`) and are optional, it's better to no set something in a [json as null](https://golang.org/pkg/encoding/json/#:~:text=The%20JSON%20null%20value%20unmarshals,value%20and%20produces%20no%20error.) (we got some side effects on other providers)

Reply:
Ah I see, thanks for the explanation